src/solana/index.ts

```diff
@@ -0,0 +1,378 @@
+   * @param stakeAccountPublicKey - The public key of the stake account.
+   * @param stakeBalance - The amount to be withdrawn from the stake account.
+   * @throws Throws an error if the token is invalid or if there's an issue during the withdrawal process.
+   * @returns Returns a promise that resolves with the withdrawal transaction.
+   *
+   */
+  public async withdraw(
+    token: string,
+    address: string,
+    stakeAccountPublicKey: PublicKey,
+    stakeBalance: number,
+  ): Promise<ApiResponse<VersionedTransaction>> {
+    // Check if the token is valid
+    const isTokenValid = await CheckToken(token);
+    if (!isTokenValid) {
+      this.throwError('INVALID_TOKEN_ERROR');
+    }
+
+    try {
+      const publicKey = new PublicKey(address);
+      const stakeAccount = new PublicKey(stakeAccountPublicKey);
+
+      // Create the withdraw transaction
+      const withdrawTx = new Transaction().add(
+        StakeProgram.withdraw({
+          stakePubkey: stakeAccount,
+          authorizedPubkey: publicKey,
+          toPubkey: publicKey,
+          lamports: stakeBalance,
+        }),
+      );
+
+      // Update the stats
+      await SetStats({
+        token,
+        action: 'unstake',
+        amount: stakeBalance / LAMPORTS_PER_SOL,
+        address,
+        chain: CHAIN,
+      });
+
+      const withdrawVerTx = await this.prepareTransaction(
+        withdrawTx.instructions,
+        publicKey,
+      );
+
+      return { result: withdrawVerTx };
+    } catch (error) {
+      throw this.handleError('WITHDRAW_ERROR', error);
+    }
+  }
+
+  /**
+   * Fetches the delegations of a given account.
+   *
+   * @param address - The public key of the account.
+   * @throws Throws an error if there's an issue fetching the delegations.
+   * @returns Returns a promise that resolves with the delegations of the account.
+   *
+   */
+  public async getDelegations(
+    address: string,
+  ): Promise<ApiResponse<Array<Delegation>>> {
+    try {
+      // Define the stake program address
+      const stakeProgramAddress = new PublicKey(
+        'Stake11111111111111111111111111111111111111',
+      );
+
+      // Define the filters for the getParsedProgramAccounts method
+      const filters = [
+        { dataSize: 200 },
+        { memcmp: { offset: 44, bytes: address } },
+      ];
+
+      // Fetch the accounts
+      const accounts = await this.connection.getParsedProgramAccounts(
+        stakeProgramAddress,
+        { filters },
+      );
+
+      return { result: accounts };
+    } catch (error) {
+      throw this.handleError('GET_DELEGATIONS_ERROR', error);
+    }
+  }
+
+  /**
+   * Stakes a certain amount of lamports.
+   *
+   * @param sender - The public key of the sender.
+   * @param lamports - The number of lamports to stake.
+   *
+   * @returns A promise that resolves to a VersionedTransaction object.
+   */
+  async stake(
+    sender: string,
+    lamports: number,
```

Comment:
And here lamports are used for the amount. I'd suggest using lamports/token units for any amount the SDK accepts for consistency.